SDWebImage/SDWebImageDownloaderOperation.m

```diff
@@ -105,10 +105,15 @@ - (nullable id)addHandlersForProgress:(nullable SDWebImageDownloaderProgressBloc
 
 - (nullable NSArray<id> *)callbacksForKey:(NSString *)key {
     LOCK(self.callbacksLock);
-    NSMutableArray<id> *callbacks = [[self.callbackBlocks valueForKey:key] mutableCopy];
+    NSArray<SDCallbacksDictionary *> *callbackBlocks = [self.callbackBlocks copy];
     UNLOCK(self.callbacksLock);
-    // We need to remove [NSNull null] because there might not always be a progress block for each callback
-    [callbacks removeObjectIdenticalTo:[NSNull null]];
+    NSMutableArray<id> *callbacks = [NSMutableArray arrayWithCapacity:callbackBlocks.count];
+    for (SDCallbacksDictionary *callbacksDic in callbackBlocks) {
+        id callback = callbacksDic[key];
+        if (callback) {
+            [callbacks addObject:callback];
+        }
+    }
```

Comment:
I don't necessarily like this approach. It's too verbose. I think the old code was better in terms of readability.
Ideally we would use a `flatMap` or `reduce` here, but those are only available in Swift.